qt/scientific_interfaces/Indirect/test/IndirectDataTablePresenterTest.h

```diff
@@ -0,0 +1,374 @@
+
+    assertValueIsGlobal(START_X_COLUMN, startX);
+  }
+
+  void
+  test_that_the_setEndX_slot_will_alter_the_relevant_endX_column_in_the_table() {
+    TableItem const endX(1.1);
+
+    m_presenter->setEndX(endX.asDouble(), 0);
+
+    assertValueIsGlobal(END_X_COLUMN, endX);
+  }
+
+  void
+  test_that_the_setExcludeRegion_slot_will_alter_the_relevant_excludeRegion_column_in_the_table() {
+    TableItem const excludeRegion("2-3");
+
+    m_presenter->setExcludeRegion(excludeRegion.asString(), 0);
+
+    assertValueIsGlobal(EXCLUDE_REGION_COLUMN, excludeRegion);
+  }
+
+  void
+  test_that_setGlobalFittingRange_will_set_the_startX_and_endX_taken_from_the_fitting_range() {
+    std::size_t const index(0);
+    TableItem const startX(1.0);
+    TableItem const endX(2.0);
+    auto const range = std::make_pair(startX.asDouble(), endX.asDouble());
+
+    ON_CALL(*m_model, getFittingRange(index, 0)).WillByDefault(Return(range));
+
+    EXPECT_CALL(*m_model, getFittingRange(index, 0)).Times(1);
+
+    m_presenter->setGlobalFittingRange(true);
+
+    assertValueIsGlobal(START_X_COLUMN, startX);
+    assertValueIsGlobal(END_X_COLUMN, endX);
+  }
+
+  void
+  test_that_setGlobalFittingRange_will_set_the_excludeRegion_when_passed_true() {
+    std::size_t const index(0);
+    TableItem const excludeRegion("1-2");
+
+    ON_CALL(*m_model, getExcludeRegion(index, 0)).WillByDefault(Return("1-2"));
+
+    EXPECT_CALL(*m_model, getExcludeRegion(index, 0)).Times(1);
+
+    m_presenter->setGlobalFittingRange(true);
+
+    assertValueIsGlobal(EXCLUDE_REGION_COLUMN, excludeRegion);
+  }
+
+  void
+  test_that_setGlobalFittingRange_will_connect_the_cellChanged_signal_to_updateAllFittingRangeFrom_when_passed_true() {
+    TableItem const startX(1.0);
+
+    m_presenter->setGlobalFittingRange(true);
+    m_table->item(0, START_X_COLUMN)->setText(startX.asQString());
+
+    assertValueIsGlobal(START_X_COLUMN, startX);
+  }
+
+  void
+  test_that_setGlobalFittingRange_will_disconnect_the_cellChanged_signal_when_passed_false_so_that_startX_is_not_global() {
+    int const row(1);
+    TableItem const startX(2.5);
+
+    m_presenter->setGlobalFittingRange(false);
+    m_table->item(row, START_X_COLUMN)->setText(startX.asQString());
+
+    assertValueIsNotGlobal(row, START_X_COLUMN, startX);
+  }
+
+  void
+  test_that_setGlobalFittingRange_will_disconnect_the_cellChanged_signal_when_passed_false_so_that_endX_is_not_global() {
+    int const row(0);
+    TableItem const endX(3.5);
+
+    m_presenter->setGlobalFittingRange(false);
+    m_table->item(row, END_X_COLUMN)->setText(endX.asQString());
+
+    assertValueIsNotGlobal(row, END_X_COLUMN, endX);
+  }
+
+  void test_the_enableTable_slot_will_enable_the_table() {
+    m_presenter->disableTable();
+    TS_ASSERT(!m_table->isEnabled());
+
+    m_presenter->enableTable();
+    TS_ASSERT(m_table->isEnabled());
+  }
+
+  void test_the_disableTable_slot_will_enable_the_table() {
```

Comment:
1. test_the_disableTable_slot_will_disable_the_table ?
2. are `test_the_disableTable_slot_will_enable_the_table` and `test_the_enableTable_slot_will_enable_the_table` testing something distinctly different, since you're calling and testing `disableTable` and `enableTable` in both?

Reply:
One of them is testing that the table is re-enabled after having previously been disabled. The other one is testing that the table has been disabled after having previously been enabled. It is only a small difference but I thought it can't hurt to make sure the table is in the correct state before testing that it can be disabled (or enabled) correctly.